scripts/generate_driver_wrappers.py

```diff
@@ -178,8 +178,15 @@ def main() -> int:
 
     mbedtls_root = os.path.abspath(args.mbedtls_root)
 
+    library_dir = ''
```

Comment:
It would be better to change `mbedtls_root` to `repo_root` in this file. Then comes up the question of what do we do with:
```
def_arg_mbedtls_root = build_tree.guess_mbedtls_root()
```
In `build_tree.py`, `guess_mbedtls_root` actually detects the mbedtls or TF-PSA-Crypto root.
Probably we should have in `build_tree.py`, three functions: `guess_repo_root`, `guess_mbedtls_root` and `guess_tf_psa_crypto_root` and use `guess_repo_root` here. We would then have:
```
def_arg_repo_root = build_tree.guess_repo_root()
```
Otherwise, please look at the occurrences of `library` in the file and update the ones that have to be updated.

Reply:
Ultimately we may need different functions for when Mbed TLS has TF-PSA-crypto as a submodule: `guess_repo_root` vs `guess_crypto_root` (currently same as `guess_repo_root`). I don't think we need `guess_mbedtls_root` as part of the interface. But it can stay for a while as a transitional alias for `guess_repo_root`.

For implementation, we can have a single function that works in all cases. The goal of this function is just to find how many times to go to the ultimate parent directory, it doesn't care about the exact names of the toplevel directories. 

And maybe we should introduce functions for `crypto_core_directory` and `crypto_builtins_directory`?

Reply:
> And maybe we should introduce functions for `crypto_core_directory` and `crypto_builtins_directory`?
In build_tree.py? I am not sure what do you mean by that.

Reply:
`generate_driver_wrappers` needs to know where to put its output: that's different in mbedtls vs tpc. This kind of knowledge is the job of `build_tree`.

Reply:
Thus it would be a function in build_tree.py, let's say `crypto_core_directory` that would return the path of the `library` directory in the case of mbedtls and the path to the `core` directory in the case of TF-PSA-Crypto?

Reply:
The work on `build_tree` seems to get things a bit too far for this PR I'd say. I still feel the need to rename `mbedtls_root` to `repo_root` otherwise I would have the impression to let things in a bad state but probably ok to keep
`def_arg_repo_root = build_tree.guess_mbedtls_root()` for the time being. @tom-daubney-arm @gilles-peskine-arm what do you think?

Reply:
This PR is adding dual-repo support to `generate_driver_wrappers.py`. This involves some new directory name detection code. (`library` vs `core`). Since this detection is not specific to `generate_driver_wrappers.py`, we should add it directly to `build_tree.py` rather than add it to `generate_driver_wrappers.py` and do extra work to move it later.

Reply:
Thus, what about if we add a function `crypto_core_directory` in build_tree.py, that returns the path of the `library` directory in the case of mbedtls (directory that currently contains the PSA core code in mbedtls) and the path to the `core` directory in the case of TF-PSA-Crypto? Then, when mbedtls is restructured to host TF-PSA-Crypto in the mbedtls case the function will return the path to `core` as well.

Reply:
@ronald-cron-arm This seems sensible to me. Any further work on repository detection is being tracked in #8524 and so can be sorted out when I do that ticket, which will be immediately after this ticket. Is that ok with you @gilles-peskine-arm ?

Reply:
I've made what I think are the suggested improvements around variable naming. I may have gone a little further than was intended so I can walk it a back a bit if needed but I have scrubbed all mention of mbedtls in the variable names and functions that are local to `scripts/generate_driver_wrappers.py`.